test/functional/feature_fee_estimation.py

```diff
@@ -255,9 +257,44 @@ def run_test(self):
             self.transact_and_mine(10, self.nodes[1])
             check_estimates(self.nodes[1], self.fees_per_kb)
 
+        # Test the conf_target argument for sendtoaddress when whole coins are spent
+        # and subtractfeefromamount is used
+        low_count = 288
+        high_count = 2
+        low_feerate = self.nodes[1].estimatesmartfee(low_count, "ECONOMICAL")
+        high_feerate = self.nodes[1].estimatesmartfee(high_count, "ECONOMICAL")
+        assert high_feerate["feerate"]/2 > low_feerate["feerate"] # comfortably larger
+        assert high_feerate["blocks"] < low_feerate["blocks"]
+
+        # Spend all coins at once
+        self.nodes[1].sendtoaddress(address=self.nodes[1].getnewaddress(), amount=self.nodes[1].getbalance(), subtractfeefromamount=True, conf_target=low_count, estimate_mode="ECONOMICAL")
+        self.nodes[1].generatetoaddress(1, ADDRESS_BCRT1_UNSPENDABLE)
+        # Then re-spend the change a couple times to test close as possible scenarios
+        low_txid = self.nodes[1].sendtoaddress(address=self.nodes[1].getnewaddress(), amount=self.nodes[1].getbalance(), subtractfeefromamount=True, conf_target=low_count, estimate_mode="ECONOMICAL")
+        high_txid = self.nodes[1].sendtoaddress(address=self.nodes[1].getnewaddress(), amount=self.nodes[1].getbalance(), subtractfeefromamount=True, conf_target=high_count, estimate_mode="ECONOMICAL")
+        # Send exactly the UTXO amount to multiple destinations, with one output paying the fee
+        sum_total = self.nodes[1].getbalances()["mine"]["trusted"]
+        subtract_address = self.nodes[1].getnewaddress()
+        low_many_txid = self.nodes[1].sendmany(amounts={self.nodes[1].getnewaddress(): sum_total-10, subtract_address: 10}, subtractfeefrom=[subtract_address], conf_target=low_count, estimate_mode="ECONOMICAL")
+        sum_total = self.nodes[1].getbalances()["mine"]["trusted"]
+        high_many_txid = self.nodes[1].sendmany(amounts={self.nodes[1].getnewaddress(): sum_total-10, subtract_address: 10}, subtractfeefrom=[subtract_address], conf_target=high_count, estimate_mode="ECONOMICAL")
+
+
+        # Total fee should be really close to one-off estimates
+        low_info = self.nodes[1].getmempoolentry(low_txid)
+        high_info = self.nodes[1].getmempoolentry(high_txid)
+        low_many_info = self.nodes[1].getmempoolentry(low_many_txid)
+        high_many_info = self.nodes[1].getmempoolentry(high_many_txid)
+
+        assert_approx(low_info["vsize"]*low_feerate["feerate"]/1000, float(low_info["fee"]))
+        assert_approx(high_info["vsize"]*high_feerate["feerate"]/1000, float(high_info["fee"]))
+        assert_approx(low_many_info["vsize"]*low_feerate["feerate"]/1000, float(low_many_info["fee"]))
```

Comment:
fails

https://travis-ci.org/github/bitcoin/bitcoin/jobs/716915333#L8855

Reply:
hmm swore I ran it locally, looks like it's *just* outside the approx range. I'll do some log-hunting to make sure nothing is especially weird and relax the range a bit if not.

Reply:
Seems like a really internal piece of fee estimation machinery difference that only happens every other time:

```
2020-08-31T13:47:22.928489Z [httpworker.3] [default wallet] Fee Calculation: Fee:17062 Bytes:208 Needed:17062 Tgt:2 (requested 2) Reason:"Double Target 95% Threshold" Decay 0.96200: Estimation: (80730.4 - 93455.5) 97.69% 28.8/(29.4 0 mem 0.0 out) Fail: (76886.1 - 80730.4) 93.58% 39.6/(42.3 0 mem 0.0 out)
```

vs

```
 node1 2020-08-11T14:14:23.348964Z [default wallet] Fee Calculation: Fee:23514 Bytes:208 Needed:23514 Tgt:2 (requested 2) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (108186 - 113596) 91.18% 38.1/(41.8 0 mem 0.0 out) Fail: (93455.5 - 108186) 80.72% 50.3/(62.4 0 mem 0.0 out) 
```